File: Day 62/Solution.java
// 58. Length of Last Word
// Given a string s consisting of words and spaces, return the length of the last word in the string.

// A word is a maximal substring consisting of non-space characters only.
public class Solution {
    public int lengthOfLastWord(String s) {
        s = s.trim(); // Remove trailing and leading spaces
        String[] words = s.split(" ");
        return words[words.length - 1].length();
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.lengthOfLastWord("Hello World"));          // 5
        System.out.println(sol.lengthOfLastWord("   fly me   to   the moon  ")); // 4
        System.out.println(sol.lengthOfLastWord("luffy is still joyboy")); // 6
        System.out.println(sol.lengthOfLastWord("   ")); // 0 or error depending on problem constraints
    }
}

File: Day 62/Solution.py
# 58. Length of Last Word
# Given a string s consisting of words and spaces, return the length of the last word in the string.

# A word is a maximal substring consisting of non-space characters only.
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        return len(s.rstrip().split()[-1]) if s.strip() else 0

# Test cases
if __name__ == "__main__":
    s = Solution()
    print(s.lengthOfLastWord("Hello World"))          # 5
    print(s.lengthOfLastWord("   fly me   to   the moon  ")) # 4
    print(s.lengthOfLastWord("luffy is still joyboy")) # 6
    print(s.lengthOfLastWord("   "))                  # 0
